### PR TITLE
VMware: Fix vmware_vm_inventory

### LINKED ISSUE #50249: vmware_vm_inventory lacks necessary information
##### SUMMARY
The `vmware_vm_inventory` plugin does not appear to try and set any useful hostvars for hosts it creates.

At minimum, this inventory should set as much useful information as possible, somewhat similar to the return data you get from `vmware_guest` and other modules or even the original contrib inventory scripts for `vmware.py` or `vmware_inventory.py`.

At the moment, you are given a concatenated hostname + uuid, and groups.  But no useful information for actually connecting to the host such as `ansible_host`.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
lib/ansible/plugins/inventory/vmware_vm_inventory.py

##### ANSIBLE VERSION

```paste below
2.6
2.7
2.8
```

##### CONFIGURATION

```paste below
NA
```

##### STEPS TO REPRODUCE

```yaml

```

##### EXPECTED RESULTS
Useful hostvars for connecting to the vmware guests

##### ACTUAL RESULTS
No useful hostvars for connecting to the vmware guests

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
* Added documentation around using vmware dynamic inventory plugin
* Fixed bug for populating host_ip in hostvars for given inventory host

Fixes: #50249

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
changelogs/fragments/50249-vmware_vm_inventory-fix_host_ip.yaml
docs/docsite/rst/vmware/index.rst
docs/docsite/rst/vmware/vmware_inventory.rst
lib/ansible/plugins/inventory/vmware_vm_inventory.py

### CODE REVIEW

File: changelogs/fragments/50249-vmware_vm_inventory-fix_host_ip.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+- Added documentation aobut using VMware dynamic inventory plugin.
```
Comment (samccann) [inline changelogs/fragments/50249-vmware_vm_inventory-fix_host_ip.yaml]:
```suggestion
- Added documentation about using VMware dynamic inventory plugin.
```

**Comment by Akasurde** [comment]
cc @sivel Could you please take a look ?

**Review by jctanner — APPROVED** [review]

**Comment by acozine** [comment]
Docs components look good.

**Comment by sivel** [comment]
This looks good to me:

```
            "tower_564d8443-7474-2286-e5e1-e51df940c5cf": {
                "ansible_host": "172.16.0.117",
                "config.cpuHotAddEnabled": false,
                "config.cpuHotRemoveEnabled": false,
                "config.hardware.numCPU": 2,
                "config.instanceUuid": "524c8e4b-4d39-dac1-c8cb-d8b0a93a177b",
                "config.name": "tower",
                "config.template": false,
                "customValue": [],
                "guest.guestId": "centos7_64Guest",
                "guest.guestState": "running",
                "guest.hostName": "tower",
                "guest.ipAddress": "172.16.0.117",
                "name": "tower",
                "runtime.maxMemoryUsage": 2048
            }
```

**Review by sivel — APPROVED** [review]

**Comment by Akasurde** [comment]
@sivel @acozine @jctanner @samccann Thanks for the review.

**Pull request merged**